File: frontend/src/lib/asx200.ts
export const ASX200_TICKERS = [
  "A2M","AAC","ABC","ABP","ACF","AFI","AGE","AGY","AIS","ALD",
  "ALL","ALU","ALX","AMP","AMC","ANN","ANZ","APA","APM","APX",
  "ARF","ARG","ARU","ASX","AUB","AWC","AX1","AZJ","BEN","BGA",
  "BHP","BKI","BKY","BLD","BMN","BOE","BOQ","BPT","BRG","BSL",
  "BXB","BWP","CAR","CBA","CCX","CEN","CGC","CGF","CHC","CIA",
  "CIM","CLW","COH","COL","CPU","CQR","CRN","CSL","CSR","CTD",
  "CVN","CWN","DEG","DFG","DHG","DOW","DRR","DUI","DXS","DYL",
  "EBO","EDV","ELD","ERA","EVN","FLT","FMG","FPH","GDG","GNC",
  "GOR","GPT","GQG","GWA","HCW","HDN","HLS","HMC","HUB","HUO",
  "HVN","IAG","IEL","IFL","IGO","ILU","IMU","ING","IPH","IPL",
  "JBH","JDO","JHX","KAR","LIC","LNK","LOT","LOV","LRK","LTR",
  "LYC","MFF","MFG","MGR","MIN","MPL","MQG","MYR","NAB","NAN",
  "NCM","NHC","NHF","NIC","NST","NUF","NWL","NWS","NXT","ORA",
  "ORG","ORI","OZL","PAN","PDN","PEN","PIC","PLS","PME","PMV",
  "PPT","PRN","PTM","PXA","QAN","QBE","REA","RED","REH","RGN",
  "RHC","RIO","RMD","RWC","S32","SBM","SCG","SDF","SDR","SEK",
  "SFR","SGM","SGP","SHL","SLC","SLX","SOL","STO","SUL","SUN",
  "SVW","SYR","TAH","TER","TGR","TLX","TLS","TNE","TWE","TCL",
  "TYR","VCX","VEA","VMY","WAM","WBC","WDS","WEB","WES","WHC",
  "WLE","WOW","WSA","WTC","XRO","YAL","29M",
] as const;

export type ASXTicker = (typeof ASX200_TICKERS)[number];

File: asx200.py
"""ASX 200 ticker validation list. Updated periodically."""

# Top ~200 ASX-listed companies by market cap.
# This is a static validation list — not a live index feed.
ASX200_TICKERS = frozenset({
    # Top 20
    "BHP", "CBA", "CSL", "NAB", "WBC", "ANZ", "WES", "MQG", "FMG", "WDS",
    "TLS", "WOW", "RIO", "ALL", "GMG", "TCL", "COL", "STO", "QBE", "REA",
    # 21-50
    "NCM", "AMC", "SHL", "JHX", "SOL", "ORG", "IAG", "MIN", "S32", "SUN",
    "BXB", "APA", "RMD", "CPU", "TWE", "ORI", "AZJ", "BSL", "SVW", "GPT",
    "NST", "MGR", "DXS", "CHC", "SGP", "VCX", "SCG", "ABP", "CWN", "EVN",
    # 51-100
    "ILU", "WHC", "ALD", "LYC", "PLS", "IGO", "SFR", "29M", "DEG", "GOR",
    "RED", "PRN", "BPT", "AWC", "NHC", "YAL", "CRN", "HVN", "TAH", "SGM",
    "IEL", "ASX", "MPL", "NXT", "ALX", "CEN", "DRR", "CIA", "BOQ", "BEN",
    "HUB", "NWS", "SEK", "CAR", "DHG", "REH", "WTC", "XRO", "TNE", "ALU",
    "PME", "APX", "TYR", "LNK", "PPT", "CGF", "AMP", "IFL", "MFG", "PTM",
    # 101-150
    "PDN", "ERA", "LOT", "BMN", "DYL", "PEN", "AGE", "BOE", "BKY", "VMY",
    "CCX", "OZL", "NIC", "WSA", "PAN", "LTR", "SYR", "TLX", "AGY", "ARU",
    "GQG", "HMC", "JDO", "SDF", "NWL", "QAN", "FLT", "WEB", "CTD", "CVN",
    "KAR", "BRG", "SUL", "JBH", "HVN", "PMV", "MYR", "LOV", "AX1", "SBM",
    "EDV", "RWC", "GWA", "CSR", "ABC", "BLD", "CIM", "DOW", "IPL", "NUF",
    # 151-200
    "ELD", "GNC", "CGC", "AAC", "TGR", "HUO", "SLC", "A2M", "BGA", "ING",
    "RGN", "CLW", "BWP", "CQR", "HDN", "ARF", "HCW", "GDG", "AUB", "SLX",
    "NHF", "HLS", "RHC", "ANN", "COH", "FPH", "EBO", "API", "TER", "IMU",
    "LIC", "WAM", "AFI", "ARG", "MFF", "WLE", "PIC", "ACF", "BKI", "DUI",
    "VEA", "AMP", "ORA", "IPH", "NAN", "PXA", "LRK", "AIS", "SDR", "APM",
})


def is_valid_asx_ticker(ticker: str) -> bool:
    """Check if a ticker is in the ASX 200 list."""
    return ticker.upper() in ASX200_TICKERS
